adding Kommander-specific challenges for upgrading (#3149)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.1/upgrade/index.md

enterprise: false
---

Kommander is an addon running on a Konvoy cluster, it will be updated in the same way every [Konvoy addon is upgraded][addons-upgrade].

When upgrading Kommander, it is recommended to use the equivalent [new version of konvoy CLI and update][konvoy-upgrade] your `cluster.yaml` file according to the instructions.

## Kommander cluster creation before Konvoy v1.4.3

> In versions before Konvoy v1.4.3 (where Kommander version is v1.0), new Kommander clusters cannot be created. It is recommended to upgrade to the latest version of Konvoy (v1.5.0), which includes Kommander 1.1.0 to be able to create new Kommander clusters.

Reviewer: I think this is better suited for a "known issues" section in the release notes.

Author: What are your thoughts on that, @samvantran ? 

In a way, I agree, because it is a known issue, but on the other hand ... it is for anything with Kommander and only affects upgrades. I suppose we could also put something on upgrades and say "click here to see the known issues with upgrading" - and then link to release notes?

Reviewer 2: I agree with Casie. We do this with Konvoy https://docs.d2iq.com/ksphere/konvoy/1.5/release-notes/#known-issues-and-limitations so it makes sense to put those limitations in the versions that are affected.

Author: Thanks for the feedback @cmastr @samvantran 
Commit coming up.

## Kommander cluster limitations

> Currently, when you upgrade Konvoy with Kommander, you can create Kommander clusters from the pre-upgraded version, and those clusters are available and can be managed from the upgraded version of Konvoy and Kommander. However, users cannot delete the cluster from the Kommander UI, after the Konvoy version was upgraded. The version of Konvoy, on the Kommander-created cluster, will also not upgrade after the user upgrades the Konvoy CLI. Users can not upgrade Kubernetes to a version greater than what was initially available when the cluster was created, in previous versions.

Author: ```suggestion
Currently, when you upgrade Konvoy with Kommander, you can create Kommander clusters from the pre-upgraded version, and those clusters are available and can be managed from the upgraded version of Konvoy and Kommander. However, users cannot delete the cluster from the Kommander UI, after the Konvoy version was upgraded. The version of Konvoy, on the Kommander-created cluster, will not upgrade after the user upgrades the Konvoy CLI. Users can not upgrade Kubernetes to a version greater than what was initially available when the cluster was created, in previous versions.
```

> Currently, when you upgrade Konvoy with Kommander, you can create Kommander clusters from the pre-upgraded version, and those clusters will be available and be able to be managed from the upgraded version of Konvoy and Kommander. However, these users cannot delete the cluster from the Kommander UI after the Konvoy version was upgraded. The version of Konvoy on the Kommander-created cluster will also not upgrade after the user upgrades the Konvoy CLI. Users will not be able to upgrade the version of Kubernetes greater than what was initially available when the cluster was created in previous versions.

Reviewer 3: ```suggestion
Currently, when you upgrade Konvoy with Kommander, you can create Kommander clusters from the pre-upgraded version, and those clusters are available and can be managed from the upgraded version of Konvoy and Kommander. However, users cannot delete the cluster from the Kommander UI, after the Konvoy version was upgraded. The version of Konvoy, on the Kommander-created cluster, will also not upgrade after the user upgrades the Konvoy CLI. Users can not upgrade Kubernetes to a version greater than what was initially available when the cluster was created, in previous versions.
```
